src/pages/signin/PasswordForm.js

```diff
@@ -35,28 +42,57 @@ const propTypes = {
 };
 
 const defaultProps = {
-    account: {},
+    account: {
+        password: '',
+        twoFactorAuthCode: '',
+    },
 };
 
 class PasswordForm extends React.Component {
     constructor(props) {
         super(props);
 
+        this.updatePassword = this.updatePassword.bind(this);
+        this.updateTwoFactorAuthCode = this.updateTwoFactorAuthCode.bind(this);
         this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
+        this.password = props.account.password;
+        this.twoFactorAuthCode = props.account.twoFactorAuthCode;
 
         this.state = {
             formError: false,
-            password: '',
-            twoFactorAuthCode: '',
         };
     }
 
+    componentWillUnmount() {
+        Onyx.merge(ONYXKEYS.ACCOUNT, {password: '', twoFactorAuthCode: ''});
```

Comment:
While testing this I realized we call `componentWillUnmount` when switching from the Wide to Narrow view (or visa versa) so we can't actually call this here.  Because of this, I think it's sufficient to call it in `validateAndSubmitForm()`, though we could alternatively add `componentWillUnmount()` to `SigninPage.js`

Reply:
>  we could alternatively add componentWillUnmount() to SigninPage.js

I'd like to follow this approach if it's alright.

Reply:
Sure, that's fine